deploy/Dockerfile

```diff
@@ -18,13 +18,12 @@ FROM golang:1.10
 WORKDIR /go/src/github.com/GoogleContainerTools/kaniko
 COPY . .
 RUN make
-WORKDIR /usr/local/bin
-ADD https://github.com/GoogleCloudPlatform/docker-credential-gcr/releases/download/v1.4.3-static/docker-credential-gcr_linux_amd64-1.4.3.tar.gz .
-RUN tar -xvzf /usr/local/bin/docker-credential-gcr_linux_amd64-1.4.3.tar.gz
+# Get GCR credential helper
+ADD https://github.com/GoogleCloudPlatform/docker-credential-gcr/releases/download/v1.4.3-static/docker-credential-gcr_linux_amd64-1.4.3.tar.gz /usr/local/bin/
```

Comment:
Hmm, shouldn't adding the tar also handle the unpacking below? It would be nice to unpack in the same step we add, to avoid having the .tar.gz in the final image.

Reply:
[ADD](https://docs.docker.com/engine/reference/builder/#add) only unpacks local tar archives, so we need to unpack in the next step. The .tar.gz shouldn't end up in the final image though, because I only copy over the unpacked binary in the next stage

Reply:
lol... thanks for explaining